tests/test_google_sheets_datatypes.py

```diff
@@ -0,0 +1,278 @@
+        # Verify that the sheet's sheet_metadata column types are consistent with the sheet's schema
+        for column_name in schema_column_names:
+            with self.subTest(column=column_name):
+                column_type = column_name_to_type[column_name]
+
+                expected_schema = google_datatypes_to_json_schema[column_type]
+                actual_schema = test_sheet_schema['properties'][column_name]
+
+                self.assertDictEqual(expected_schema, actual_schema)
+
+        ##########################################################################
+        ### Test the happy path datatype values
+        ##########################################################################
+
+        # Verify that all data has been coerced to the expected column datatype
+        record_data = [message['data'] for message in synced_records[test_sheet]['messages'] if message.get('data')]
+        data_map = {
+	    "Currency": Decimal, # BUG Currency is being identified as a decimal type rather than string https://jira.talendforge.org/browse/TDL-14360
+	    "Datetime": str,
+            "Time": str,
+            "Date": str,
+	    "String": str,
+            "Number": Decimal,
+            "Boolean": bool,
+        }
+        string_column_formats = {
+	    "Datetime": "%Y-%m-%dT%H:%M:%S.%fZ",
+            "Time":  "%H:%M:%S",
+            "Date": "%Y-%m-%d",
+        }
+
+        for record in record_data:
+            with self.subTest(record=record):
+                for column in data_map.keys():
+                    with self.subTest(column=column):
+                        if record.get(column):
+
+                            # verify the datatypes for happysheet values match expectations
+                            self.assertTrue(
+                                isinstance(record[column], data_map[column]),
+                                msg=f'actual_type={type(record[column])}, expected_type={data_map[column]}, value={record[column]}'
+                            )
+
+                            # TODO implement this when requirements are clear [https://jira.talendforge.org/browse/TDL-14431]
+                            # # verify dates, times and datetimes are of the expected format
+                            # if column in string_column_formats.keys():
+                            #     self.assertStringFormat(record[column], string_column_formats[column])
+
+        ##########################################################################
+        ### Test the string fallbacks for each datatype
+        ##########################################################################
+
+        test_sheet = 'happysheet-string-fallback'
+
+        # get field names for the tested sheet frome the replicated schema
+        record_data = [message['data']
+                       for message in synced_records[test_sheet]['messages']
+                       if message.get('data')]
+        columns = set(data_map.keys())
+
```

Comment:
Need to put back the assertStringFormat above,  and write up the sting-fallback case where a Date will replicate a time as epoch.